### PR TITLE
add --kill-signal option

### LINKED ISSUE #395: Integration tests never complete
**Environment**
MacOS 13.0.1
Node 16.16.0
Typescript 4.9.4
Concurrently 7.6.0

**Describe the bug**
We are attempting to write some integration tests for our azure functions, however the function runtime restarts instead of exiting when the test are complete, so the process never finishes.

**To Reproduce**
1. Create a new function app
2. Create at least one http trigger
3. Write an integration test using `jest` which makes a http request against the trigger
4. Install the following packages: jest, ts-jest, concurrently, wait-on
5. Add the following scripts to `package.json`

```
"pretest": "tsc -p tsconfig.test.json",
"test": "concurrently -k -s last \"func start\" \"wait-on tcp:7071 && jest\"",
```

6. Execute the following command on the terminal: `npm run test`
7. Wait for process to complete.

**Expected behavior**
When the jest exits, concurrently should kill the azure function runtime, and terminate.

**Actual behavior**
1. Jest exits successfully.
2. Concurrently sends a kill signal to the azure function runtime.
3. The azure function runtime restarts itself and keeps running.
4. Integration tests never complete, because runtime is still active.

**Relevant Console Output**
```
[1] Ran all test suites.
[1] wait-on tcp:7071 && jest exited with code 0
--> Sending SIGTERM to other processes..
[0] [2022-12-08T16:06:34.042Z] Language Worker Process exited. Pid=64056.
[0] [2022-12-08T16:06:34.042Z] node exited with code 143 (0x8F). .
[0] [2022-12-08T16:06:34.280Z] Worker process started and initialized.
```

**Proposed Fix**
Add a command line argument to `concurrently` which will send a SIGKILL signal instead of a SIGTERM signal, to force the azure runtime to exit immediately.

### PR DESCRIPTION
**Pull request by paulsmithkc**
Add a --kill-signal option to change what signal is sent to the other processes.

Fixes #395

### CODE REVIEW

**Comment by paulsmithkc** [comment]
@paescuj @gustavohenke bump

**Comment by jessephelps** [comment]
This would be helpful for my issues as well!

File: bin/concurrently.ts [hunk 1]
```diff
@@ -110,6 +110,11 @@ const args = yargs(argsBeforeSep)
             describe: 'Kill other processes if one exits with non zero status code.',
             type: 'boolean',
         },
+        'kill-signal': {
+            alias: 'ks',
+            describe: 'Signal to send to other processes if one exits or dies.',
```
Comment (gustavohenke) [inline bin/concurrently.ts:115]: It'd be useful to list what the default is here. Take a look at `passthrough-arguments` and `timings` for some examples

File: src/flow-control/kill-others.spec.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ const createWithConditions = (conditions: ProcessCloseCondition[]) =>
     new KillOthers({
         logger,
         conditions,
+        killSignal: undefined,
```
Comment (gustavohenke) [inline src/flow-control/kill-others.spec.ts:26]: We need to extend the test file to cover the arguments passed to the `command.kill()`, both when `undefined` and the custom one

**Review by gustavohenke — CHANGES_REQUESTED** [review]

File: bin/concurrently.ts (hunk 1 above)
Comment (paulsmithkc) [reply bin/concurrently.ts]: Fixed.

File: src/flow-control/kill-others.spec.ts (hunk 2 above)
Comment (paulsmithkc) [reply src/flow-control/kill-others.spec.ts]: Fixed.

**Review by gustavohenke — APPROVED** [review]
Thanks 🎉

**Comment by gustavohenke** [comment]
Released in v8.0.0!

**Pull request merged**